File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Commands/RunExample.java
package Model.Commands;

import Controller.ToyProgramController;

public class RunExample extends Command {
    private ToyProgramController controller;

    public RunExample(String key, String description, ToyProgramController controller){
        super(key, description);
        this.controller = controller;
    }

    @Override
    public void execute() {
        try{
            System.out.println(controller.getProgram().toString());
            controller.allStep();
            System.out.println(controller.getOutput());
        }
        catch(Exception e){
            System.out.println("from RunExample" + e.toString());
        }
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Commands/RunExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageInterpreter.Src.Controller;

namespace ToyLanguageInterpreter.Src.Model.Commands
{
    class RunExample : Command
    {
        private ToyProgramController controller;

        public RunExample(string key, string description, ToyProgramController controller) : base(key, description) 
        {
            this.controller = controller;
        }

        public override void Execute()
        {
            try
            {
                Console.WriteLine(controller.ToString());
                controller.AllStep();
                Console.WriteLine(controller.Output);
            }
            catch (Exception e)
            {
                Console.WriteLine("From RunExample" + e.ToString());
            }
        }
    }
}
